### ISSUE TITLE
Incorrect assumption: "0x41 is 'C'"

### ISSUE THREAD

**Issue by ariccio** [google/UIforETW#9]
[I saw the comment at line 261:](https://github.com/google/UIforETW/blob/master/UIforETW/UIforETWDlg.cpp#L261)

```
	// 0x41 is 'C', compatible with wprui
```
So I figured that a `static_assert` was in order:

```
static_assert( 'C' == 0x41, "Invalid assumption!" );
```
...and it fails.

Typing:
```
const DWORD c = 'C';
```
...intellisense informs me that `c == 67UL`.

What's up with that?

**Comment by randomascii** [comment]
The 0x41 part of the comment is wrong (note that it doesn't match the code). It should say 0x43, or else the 0x43 parameter should be changed to 'C'.

Comment should probably be:

// Ctrl+Win+C, compatible with wprui's trace recording hot key

The goal is to be able to record traces with Ctrl+Win+C, same as wprui.

**Comment by ariccio** [comment]
Or, (maybe) better yet, instead of:

```
    // 0x41 is 'C', compatible with wprui
```

this:

```
const UINT hotKeyCode = 0x43;
static_assert( 'C' == hotKeyCode, "This key code is NOT compatible with WPRUI's trace recording hot key"
// wprui's trace recording hot key combo is: Ctrl+Win+C, we want to mimic this ability.
if (!RegisterHotKey(*this, kRecordTraceHotKey, MOD_WIN + MOD_CONTROL, hotKeyCode))
```

...so that if someone ever changes the hardcoded hotkey code, they get an informative error code, instead of a misbehaving app?

**Comment by randomascii** [comment]
I don't see what the static assert adds. Just set hotKeyCode to 'C' if
that's what it is supposed to be.

If somebody wants to change the hot key they can do that and the only
invariant that they should maintain is that the text on the button should
match the new hot key, but statically asserting that is a bit clumsy.

On Wed, Apr 15, 2015 at 8:23 PM, Alexander Riccio <notifications@github.com>
wrote:

> Or, (maybe) better yet, instead of:
>
>     // 0x41 is 'C', compatible with wprui
>
> this:
>
> const UINT hotKeyCode = 0x43;
> static_assert( 'C' == hotKeyCode, "This key code is NOT compatible with WPRUI's trace recording hot key"
> // wprui's trace recording hot key combo is: Ctrl+Win+C, we want to mimic this ability.
> if (!RegisterHotKey(*this, kRecordTraceHotKey, MOD_WIN + MOD_CONTROL, hotKeyCode))
>
> ...so that if someone ever changes the hardcoded hotkey code, they get an
> informative error code, instead of a misbehaving app?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/UIforETW/issues/9#issuecomment-93627667>.
>



-- 
Bruce Dawson

**Comment by randomascii** [comment]
Fixed with change 55d092751e8b241bd2ea820dd9ef8f2a14392b4b

**Comment by ariccio** [comment]
Yeah, it's really just nitpicking on my part.
